app/assets/javascripts/discourse/app/components/user-notification-schedule.js

```diff
@@ -0,0 +1,121 @@
+import Component from "@ember/component";
+import I18n from "I18n";
+import { action } from "@ember/object";
+import discourseComputed from "discourse-common/utils/decorators";
+
+export default Component.extend({
+  startTimeOptions: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.set(
+      "startTimeOptions",
+      this.buildTimeOptions(0, {
+        includeNone: true,
+        showMidnight: false,
+      })
+    );
+  },
+
+  buildTimeOptions(startAt, opts = { includeNone: false, showMidnight: true }) {
+    let timeOptions = [];
+
+    if (opts.includeNone) {
+      timeOptions.push({
+        name: I18n.t("user.notification_schedule.none"),
+        value: -1,
+      });
+    }
+
+    for (let timeInMin = startAt; timeInMin <= 1440; timeInMin += 30) {
+      let hours = Math.floor(timeInMin / 60);
+      let minutes = timeInMin % 60;
+      let am = timeInMin < 720;
+      if (minutes === 0) {
+        minutes = "00";
+      }
+      if (hours === 0) {
+        hours = "12";
+      }
+      if (hours === 24) {
+        if (opts.showMidnight) {
+          timeOptions.push({
+            name: I18n.t("user.notification_schedule.midnight"),
+            value: 1440,
+          });
+        }
+        break;
+      }
+      if (!am && hours !== 12) {
+        hours = hours - 12;
+      }
+      timeOptions.push({
+        name: `${hours}:${minutes} ${am ? "AM" : "PM"}`,
```

Comment:
I know it's v1, but AM/PM support is locale dependant so we should have this hardcoded.